src/components/ViewClub.js

```diff
@@ -0,0 +1,361 @@
+/* eslint-disable no-unused-vars */
+import React, { Component } from 'react';
+import {
+    CLUB_ALREADY_EXISTS, UNEXPECTED_ERROR, RESOURCE_NOT_FOUND, CLUB_UPDATED, INVALID_INFO
+} from '../constants';
+import {
+    Grid,
+    Row,
+    Col,
+    Button,
+    Label,
+    Glyphicon,
+    Alert,
+    Badge,
+    PageHeader,
+    ButtonToolbar,
+    ButtonGroup,
+    FormGroup
+} from 'react-bootstrap';
+/* eslint-enable no-unused-vars */
+
+class ViewClub extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            name: '',
+            description: '',
+            websiteUrl: '',
+            facebookUrl: '',
+            instagramUrl: '',
+            twitterUrl: '',
+            memberships: [],
+            isEditable: false,
+            isEditing: false,
+            msg: undefined,
+            isSuccess: undefined,
+        };
+
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleInput = this.handleInput.bind(this);
+        this.renderAsForm = this.renderAsForm.bind(this);
+        this.renderAsUneditable = this.renderAsUneditable.bind(this);
+    }
+
+    /**
+     * Fetches club info after the page has rendered.
+     */
+    componentDidMount() {
+        // Fetch the club's info after rendering
+        this.props.client.getClub(this.props.name)
+            .then(response => {
+                if (response.ok) {
+                    response.json().then(body => {
+                        this.setState({
+                            name: body.name,
+                            description: body.description,
+                            websiteUrl: body.websiteUrl,
+                            facebookUrl: body.facebookUrl,
+                            instagramUrl: body.instagramUrl,
+                            twitterUrl: body.twitterUrl,
+                        });
+                    });
+                } else if (response.status === 404) {
+                    this.setState({
```

Comment:
you can put it on one line so its easier to read. This function takes up a lot of space room to read the code